internal/proxy/validate_util.go

```diff
@@ -518,24 +523,24 @@ func (v *validateUtil) checkArrayFieldData(field *schemapb.FieldData, fieldSchem
 		if err != nil {
 			return err
 		}
-		for _, row := range data.GetData() {
-			if err := verifyLengthPerRow(row.GetStringData().GetData(), maxLength); err != nil {
-				return err
+		for rowCnt, row := range data.GetData() {
+			if i, ok := verifyLengthPerRow(row.GetStringData().GetData(), maxLength); !ok {
+				return merr.WrapErrParameterInvalidMsg("the length (%d) of %dth %s %s[%d] exceeds max length (%d)",
+					len(row.GetStringData().GetData()[i]), rowCnt, fieldSchema.GetDataType().String(), fieldSchema.GetName(), i, maxLength)
 			}
 		}
 	}
 	return v.checkArrayElement(data, fieldSchema)
 }
 
-func verifyLengthPerRow[E interface{ ~string | ~[]byte }](strArr []E, maxLength int64) error {
+func verifyLengthPerRow[E interface{ ~string | ~[]byte }](strArr []E, maxLength int64) (int, bool) {
```

Comment:
Any particular reason to change the method signature here?

Reply:
IMHO, to return the error index